camel/functions/openai_function.py

```diff
@@ -89,6 +90,11 @@ def get_openai_tool_schema(func: Callable) -> Dict[str, Any]:
         param_default = p.default
         param_kind = p.kind
         param_annotation = p.annotation
+        # Output like Tuple[str,str] from pydantic create_model() is not
+        # suppored by openai request, convert into Tuple
+        if isinstance(param_type,
+                      _GenericAlias) and param_type.__origin__ is tuple:
+            param_type = Tuple
```

Comment:
As I test, the output for `tuple` would be something looks like

```python
'a': {'maxItems': 2,
      'minItems': 2,
      'prefixItems': [{'type': 'string'}, {'type': 'string'}],
      'type': 'array'}
```
Here `a` is a variable annotated as `Tuple[str, str]`. If change `str` to `float`. Just the type would be changed to `float`.

As a comparison, `b` is `List[str]`

```python
'b': {'items': {'type': 'string'}, 'type': 'array'}
```

I think the return is desired since you need to specify how many arguments are in a `tuple` and what are their types. It is a different thing from `list`.
In [openai docs](https://platform.openai.com/docs/api-reference/chat/create), it refers to
[this json schema docs](https://json-schema.org/understanding-json-schema/reference/array). And you can see our output is correct for tuple. If openai requires something other than is format. I think it is a mistake from their side.

Reply:
Maybe you could have a look at https://github.com/camel-ai/camel/issues/495

it seems that tuple with specific type will lead to error,

`openai.BadRequestError: Error code: 400 - {'error': {'message': "Invalid schema for function 'get_elevation': In context=('properties', 'lat_lng'), array schema missing items", 'type': 'invalid_request_error', 'param': None, 'code': None}}`

So as you said, maybe it is a mistake from openai. @ocss884